features/feedback.feature

```diff
@@ -5,15 +5,43 @@ Feature: Feedback
     Given I am testing through the full stack
     And I force a varnish cache miss
 
+  # TODO: RENAME to clarify this is testing data transfer with
+  # Content Store (as the prime example for this app).
+  #
   Scenario: Check the "Contact GOV.UK" page loads correctly
     When I visit "/contact/govuk"
     Then I should see "Contact GOV.UK"
 
+  # TODO: EXPORT this test as it does not meet the elgibility
+  # criteria in docs/writing-tests.md.
+  #
+  # - Covers site-wide config: N (not applicable)
+  # - Targets data transfer: N (already covered)
+  # - Second critical check: N (not tested in app)
+  #
+  # Data transfer with Content Store is already covered by
+  # "Check the "Contact GOV.UK" page loads correctly".
+  #
+  # Should be tested in Feedback [^1].
```

Comment:
I suggest we change this instead of removing - feedback should test the contact form directly, but there is a path from the component at the bottom of any given page behind the "report a problem" button that uses a combination of frontend app X interfacing with the feedback form.

Reply:
1, yes. 2 though, I THINK the component builds its own form at the bottom of every page (though it might be magic ajax style shenanigans pulling through the Feedback form). Either way, it posts to the feedback app. I think there's a chance this vulnerability may be there as an independent thing, but more importantly we need to know that a page rendered by random app X can send content successfully to the feedback app, I think?

Reply:
Here we go: ["report a problem" is rendered as part of the component](https://github.com/alphagov/govuk_publishing_components/blob/main/app/views/govuk_publishing_components/components/feedback/_problem_form.html.erb).

We do have a data transfer test for one of the other API calls to feedback (below). AJAX features are admittedly a bit harder to cover elsewhere because we can't use our normal Ruby test approaches, so I think there is an argument for having a new test to cover "report a problem" specifically talking to feedback.

However, I want to focus on the tests we can remove (or export) first. The check for malicious content is part of the feedback app, so we won't lose anything if we rewrite this test there - if the protection is already being shared with the "report a problem" API then it will still be covered, implicitly or explicitly.

I think that covers everything - quite a complex discussion.

The key question: is there a reason we shouldn't EXPORT this test to feedback?

Happy to chat if there's more to discuss - might be easier at this point.

Reply:
It does make sense to export that exact test. I think there is a gap in here that could potentially be another "TO DO".